onnx/defs/tensor/defs.cc

```diff
@@ -2206,77 +2214,238 @@ ONNX_OPERATOR_SET_SCHEMA(
-            inverseIndicesTensorProto->mutable_tensor_type()->mutable_shape()->add_dim();
+            inverseIndicesTensorProto->mutable_tensor_type()
+                ->mutable_shape()
+                ->add_dim();
           }
 
           if (num_outputs >= 4) {
             countsTensorProto = ctx.getOutputType(3);
             updateOutputElemType(ctx, 3, TensorProto::INT64);
-            countsTensorProto->mutable_tensor_type()->mutable_shape()->add_dim();
+            countsTensorProto->mutable_tensor_type()
+                ->mutable_shape()
+                ->add_dim();
           }
 
           auto axisAttr = ctx.getAttribute("axis");
           if (!axisAttr) {
             // 'axis' is not provided. Input 'X' is flattened.
             // 'Y' is a 1-D tensor of unknown dimension.
             yTensorProto->mutable_tensor_type()->mutable_shape()->add_dim();
-          }
-          else 
-          {
+          } else {
             // 'axis' is provided.
             int axis = static_cast<int>(axisAttr->i());
 
-            // 'Y' has the same shape as 'X' except in the 'axis' dimension which is unknown.
-            const TensorShapeProto& input_shape = xTensorProto->tensor_type().shape();
+            // 'Y' has the same shape as 'X' except in the 'axis' dimension
+            // which is unknown.
+            const TensorShapeProto& input_shape =
+                xTensorProto->tensor_type().shape();
 
             for (int i = 0; i < input_shape.dim_size(); i++) {
-              auto* dim = yTensorProto->mutable_tensor_type()->mutable_shape()->add_dim();
-              if (i != axis){
+              auto* dim = yTensorProto->mutable_tensor_type()
+                              ->mutable_shape()
+                              ->add_dim();
+              if (i != axis) {
                 *dim = input_shape.dim(i);
               }
             }
           }
-      }));
+        }));
+
+static const char* GatherND_ver11_doc = R"DOC(
+Given `data` tensor of rank `r` >= 1, and `indices` tensor of rank `q` >= 1, this operator gathers 
+slices of `data` into an output tensor of rank `q + r - indices_shape[-1] - 1`.
+
+`indices` is an q-dimensional integer tensor, best thought of as a `(q-1)`-dimensional tensor of indices into `data`, 
+where each element defines a slice of `data`
+
+Some salient points about the inputs' rank and shape:
+ 
+1) r >= 1 and q >= 1 are to be honored. There is no dependency condition to be met between ranks `r` and `q`.
+
+2) The `indices_shape[-1]` should have a value between 1 (inclusive) and rank `r` (inclusive) 
+
+3) The value of `indices` need to be non-negative (i.e.) >= 0. The upper bound for each value in `indices` 
+   is the `data_shape` value along the corresponding axis (i.e.) `0 <= indices[:, :, i] < data_shape[i]`
+
+The output is computed as follows:
+
+The output tensor can be thought of as filling in the `indices` tensor with the appropriate slices of the input `data`.
```

Comment:
Change to "The output tensor is obtained by mapping each index-tuple in the  `indices` tensor to the corresponding slice of the input `data`"